robottelo/hosts.py

```diff
@@ -191,13 +192,15 @@ def os_version(self):
         return Version(version=version_string)
 
     def setup(self):
-        self.remove_katello_ca()
-        self.execute('subscription-manager clean')
+        if not self.blank:
+            self.remove_katello_ca()
+            self.execute('subscription-manager clean')
 
     def teardown(self):
-        if self.nailgun_host:
-            self.nailgun_host.delete()
-        self.unregister()
+        if not self.blank:
+            if self.nailgun_host:
+                self.nailgun_host.delete()
+            self.unregister()
```

Comment:
If you haven't rebased already, please do so as this method has changed and you may not need to modify this method.

Reply:
I need to keep the `self.blank` attribute there because I am trying to teardown VM that has FQDN `.` (dot). The dot is there as it is the shortest valid FQDN. It is there because the broker requires AAP to provide the `hostname` of the VM deployed. In my case the VM is completely blank - no system installed on it, and it is stopped - not running at the time it is checked out.
I am open to discussions here with you @JacobCallahan and @mshriver. This is not a thing that we should address in this PR.

Reply:
I have rebased the branch. Can you please re-review this part?

Reply:
@ogajduse i think if you just set `self.subscribed = False` on the instance, you will achieve the same result but easier.

Reply:
Not quite. `self.subscribed` is a property that `execute`s `subscription-manager register`. Running establishing an SSH session is exactly what I want to avoid.

Reply:
I was going to recommend a different way, but that's something that can be revisited later. So no worries for this PR